config/methods.config

```diff
@@ -79,14 +93,39 @@ methods {
         }
     }
 
+    sanitize_string = { raw ->
+        if (![String, GString].any{ raw in it }) {
+            throw new Exception("Input to sanitize is either empty or not a string! Provide a non-empty string.")
+            }
+        def disallowed_characters = /[^a-zA-Z\d\/_.-]/
+        return raw.replaceAll(disallowed_characters, '').replace('_', '-')
+    }
+
+    get_ids_from_bams = {
+        params.samples_to_process = [] as Set
+        params.input.each { k, v ->
+            v.each { sampleMap ->
+                def bam_path = sampleMap['BAM']
+                def bam_header = bam_parser.parse_bam_header(bam_path)
+                def sm_tags = bam_header['read_group'].collect{ it['SM'] }.unique()
+                if (sm_tags.size() > 1) {
+                    throw new Exception("${bam_path} contains multiple samples! Please run pipeline with single sample BAMs.")
+                }
+                sm_tag = methods.sanitize_string(sm_tags[0])
+                params.samples_to_process.add(['orig-id': sm_tags[0], 'id': sm_tag, 'path': bam_path, 'sample_type': k])
+            }
+        }
+    }
+
```

Comment:
`orig-id` is needed to rename the samples in the output `mutect2` VCFs from original to sanitized.  The other tools all use `TUMOR` and`NORMAL`.